padl/transforms.py

```diff
@@ -508,6 +514,32 @@ def f(x):
 
         return graph
 
+    def _pd_process_traceback(self):
+        """Find where the Transform was defined (file, lineno, file) given the traceback. """
+        a_tb = None
+        for a_tb in self._pd_traceback[::-1]:
+            if 'padl' in a_tb[0]:
+                continue
+            break
+        return f'{a_tb.filename} in {a_tb.name}\n----> {make_green(a_tb.lineno)}    {a_tb.line}'
+
+    def _pd_get_error_idx(self):
+        """Get what element of a :class:`padl.transforms.Transform` is failing if an Exception is
+        produced during an execution.
+
+        Subclasses of :class:`padl.transforms.Transform` need to implement this method.
+        """
+        return NotImplemented
+
+    def _pd_trace_error(self, position: int, arg):
+        """Add some error description to :obj:`pd_trace`. """
+        try:
+            str_ = self._pd_fullrepr(marker=(position, '\033[31m  <---- error here \033[0m'))
+            _pd_trace.append(_TraceItem(str_, self._pd_process_traceback(), arg,
+                                        self, Transform.pd_mode, position))
+        except Exception:
+            warn('Error tracing failed')
```

Comment:
Sorry one additional question. Would it be worthwhile to raise the error here in the case the trace does fail?

Reply:
if we execute this block it means that we've got an error on a transform execution, and if we raise an error here, we would show two errors in the traceback: the runtime error of the transform and the one getting _pd_trace. This could in my opinion hide a bit the main error, which is the RuntimeError got while applying.

Reply:
Ahh I see now. I thought this would swallow the error from the transform execution, but that is not the case. Okay this looks good!